modules/kinesis/src/main/scala/com/snowplowanalytics/snowplow/enrich/kinesis/Sink.scala

```diff
@@ -288,12 +292,19 @@ object Sink {
   private def putRecords(
     kinesis: AmazonKinesis,
     streamName: String,
-    records: List[PutRecordsRequestEntry]
+    records: List[PutRecordsRequestEntry],
+    maxRecordSize: Int
   ): PutRecordsResult = {
     val putRecordsRequest = {
       val prr = new PutRecordsRequest()
+      val filteredRecords = records.map { record =>
+        if (record.getData.array.length > maxRecordSize) {
+          record.setData(ByteBuffer.wrap(record.getData.array.take(maxRecordSize / 10)))
```

Comment:
This seems wrong because we are mutating the record to fit the max size but we must return a BadRow instead